src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -933,8 +951,126 @@ void *frontendservicesesion(std::string masterIP, int connFd, SQLiteDBInterface
             JasmineGraphServer *jasmineServer = new JasmineGraphServer();
             bool isSpawned = jasmineServer->spawnNewWorker(host,port,dataPort,profile,masterHost,enableNmon);
 
-        } else {
-            frontend_logger.log("Message format not recognized " + line, "error");
+        }
+        else if (line.compare(ENTITY_RESOLUTION) == 0) {
+            bool coordinatingOrg;
+            if (coordinatingOrg) {
+                int result_wr = write(connFd, SEND.c_str(), FRONTEND_COMMAND_LENGTH);
+                if(result_wr < 0) {
+                    frontend_logger.log("Error writing to socket", "error");
+                    loop = true;
+                    continue;
+                }
+                result_wr = write(connFd, "\r\n", 2);
+                if(result_wr < 0) {
+                    frontend_logger.log("Error writing to socket", "error");
+                    loop = true;
+                    continue;
+                }
+
+                // We get the name and the path to graph as a pair separated by |.
+                char graph_id[FRONTEND_DATA_LENGTH];
+                char worker_and_port[FRONTEND_DATA_LENGTH];
+                bzero(graph_id, FRONTEND_DATA_LENGTH + 1);
+
+                read(connFd, graph_id, FRONTEND_DATA_LENGTH);
+                string graphID(graph_id);
+
+                read(connFd, worker_and_port, FRONTEND_DATA_LENGTH);
+                string designatedWorkerDetails(worker_and_port);
+
+                Utils utils;
+                graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+
+                vector<string> strParams = utils.split(designatedWorkerDetails, '|');
+                string designatedHost = strParams[0];
+                string designatedPort = strParams[1];
+
+                frontend_logger.log("Graph ID received: " + graphID, "info");
+
+                if (JasmineGraphFrontEnd::graphExistsByID(graphID, sqlite)) {
+                    frontend_logger.log("Graph with ID " + graphID + " is being deleted now", "info");
+                    JasmineGraphFrontEnd::initiateEntityResolutionCoordinator(graphID, sqlite, masterIP);
+                    result_wr = write(connFd, DONE.c_str(), DONE.size());
+                    if(result_wr < 0) {
+                        frontend_logger.log("Error writing to socket", "error");
+                        loop = true;
+                        continue;
+                    }
+                    result_wr = write(connFd, "\n", 2);
+                    if(result_wr < 0) {
+                        frontend_logger.log("Error writing to socket", "error");
+                        loop = true;
+                        continue;
+                    }
+                } else {
+                    frontend_logger.log("Graph does not exist or cannot be deleted with the current hosts setting",
```

Comment:
Why do we say the graph cannot be deleted?

Reply:
Changed to log appropriate message